### DISCUSSION TITLE
How to getCurrentInstance or inject in store?

### DISCUSSION THREAD

**Discussion by tanghaojie** [vuejs/pinia#1397] [Help and Questions]
I have a container in app, like this :
```js
export default markRaw({
  install(app: App): void {
    const container = new Container(app)
    // @ts-ignore
    app.provide(CONTAINER_INJECT_KEY, container)
  },
})
```
and use  method:
```js
export function useContainer() {
  const currentInstance = getCurrentInstance()
  const container = currentInstance && inject(CONTAINER_INJECT_KEY)
  return container
}
```
then , i want to use container in store, but getCurrentInstance always get null . Is there anything wrong?

**Comment by posva** [comment]
You shouldn't be using the current instance on a store because it's not related to a component.
What are you using the instance for?

**Reply by tanghaojie** [reply]
Using the instance for inject my custom 'Container' object, when application start up.
